src/analytics/private/load-ods-data-workflow.ts

```diff
@@ -249,13 +250,18 @@ export class LoadOdsDataToRedshiftWorkflow extends Construct {
 
       const finalStatus = new Pass(this, `${odsTableName} - Job completes`);
 
+      const waitAndRetry = new Wait(this, `${odsTableName} - Wait and Retry`, {
+        time: WaitTime.duration(Duration.seconds(120)),
+      }).next(submitJob);
+
       // Create sub chain
       const subDefinition = submitJob
         .next(waitX)
         .next(checkJobStatus)
         .next(new Choice(this, `${odsTableName} - Check if job completes`)
           // Look at the "status" field
-          .when(Condition.stringEquals('$.detail.status', 'FAILED'), jobFailed)
+          .when(Condition.and(Condition.stringEquals('$.detail.status', 'FAILED'), Condition.booleanEquals('$.detail.retry', false)), jobFailed)
```

Comment:
Why not use the built-in retry of step functions?

Reply:
the failure is not for step in step functions. it is query result of redshift sql execution id.